### PR TITLE
Add support for multiple publisher threads

### PR DESCRIPTION
**Pull request by film42**
1. Add a new config flag called `publisher_threads` (default: 1).
2. Start `N` `TimerTask`s (one for each publisher thread).
3. Update specs to use the new `consumers` method (replaced `consumer`).

cc @abrandoned @quixoten

### CODE REVIEW

**Review by liveh2o — APPROVED** [review]

**Comment by film42** [comment]
Released `1.3.0.pre0`.

**Comment by film42** [comment]
@liveh2o I pushed a commit to add multiple publisher threads to the redis adapter as well.

**Comment by liveh2o** [comment]
:shipit:

**Comment by film42** [comment]
Released `1.3.0.pre1`.

**Comment by film42** [comment]
PS. This code is working just fine. But it doesn't look like it's actually solving the problem I was hoping it would solve :disappointed: .

**Comment by ReUhssurance** [comment]
We are deciding to resurrect this branch and move it forward.  I'd say LGTM, but I also added some very minor changes to instrument thread restarts, and rebased to bring this branch up to 2021 and allow for circle to actually run.

File: spec/lib/active_publisher/async/in_memory_adapter_spec.rb [hunk 1]
```diff
@@ -210,6 +242,7 @@
           verify_expectation_within(0.5) do
             expect(subject.size).to eq(0)
           end
+          sleep 1
```
Comment (ReUhssurance) [inline spec/lib/active_publisher/async/in_memory_adapter_spec.rb:246]: Jruby 9.2 exposes a race condition with the expectation on the ::ActivePublisher expectation here.

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -4,6 +4,7 @@
 require "fakeredis/rspec"
 require "active_publisher/async/redis_adapter"
 require "connection_pool"
+require "pry"
```
Comment (skunkworker) [inline spec/spec_helper.rb]: @ETetzlaff debugging leftover?

Comment (ReUhssurance) [reply spec/spec_helper.rb]: Ah, removed.

**Comment by film42** [comment]
Overall looks good to me!

Bummer to see that added `sleep 1` but I think it's good enough to test this out.

P.S. I hate to be my own reviewer but I opened this a long time ago, haha!

**Pull request merged**